### PR TITLE
Add dev log for v24.05

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: _posts/2024-05-04-openloco-v24.05.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+One of the charming features of the original Locomotion game was its use of a train for a progress bar.
+This has been missing in OpenLoco for a while, but we've brought these progress bars back in this release!
+But why were they removed in the first place?
+
+When we reimplemented loading and saving games in C++, we were suprised by how much faster our implementation was,
+such that we elected progress bars were unnecessary. However, loading objects can still take a couple of seconds,
+especially on saves that pack a bunch.
+
+In addition to progress bars for loading and saving games, the game would use 'native' Windows progress bars
+before the game is launched, e.g. to indicate it was checking and indexing its object files. We never got around to
```
Comment (duncanspumpkin) [inline _posts/2024-05-04-openloco-v24.05.md:25]: I did actually do this but it wasn't cross platform and after talking to a few people there was no good cross platform solution so i never continued

Comment (AaronVanGeffen) [reply _posts/2024-05-04-openloco-v24.05.md:25]: Ah, I thought that was for OpenRCT2! Would you like me to include a sentence for it?

Comment (duncanspumpkin) [reply _posts/2024-05-04-openloco-v24.05.md:25]: Nah it's fine

File: _posts/2024-05-04-openloco-v24.05.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+We've elected to defer number formatting to a library we had previously incorporated, libfmt.
+While we currently only support grouping numbers by thousands, we aim to support alternatives in the future,
+which should be made easier through this library.
+
+## Rework string formatting (#2461, #2464, #2469)
+
+With the last formatting codes reimplemented, we could finally refactor the way strings are reformatted.
+The previous approach was very similar to vanilla, which wasn't actually that memory safe. This mostly
+showed in to the buffer length not being taken into account at all times. This could be problematic in
```
Comment (duncanspumpkin) [inline _posts/2024-05-04-openloco-v24.05.md:116]: This sounds odd

Comment (AaronVanGeffen) [reply _posts/2024-05-04-openloco-v24.05.md:116]: Yeah, that 'to' isn't supposed to be there. Thanks.

File: _posts/2024-05-04-openloco-v24.05.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+the future, particularly when networked games come in.
+
+This month, string formatting has been completely reworked. It is now memory safe, and prepared for a
+move to unicode strings in a future release.
+
+## Rework widget allocation (#2418, #2420, #2452)
+
+On the topic of reworking code that has been with us since the project was started...
+This month, we also reworked how widgets are allocated. Previously, widgets were shared across
+all windows of the same type. Our new approach allocates widgets for the window, adding storage
```
Comment (duncanspumpkin) [inline _posts/2024-05-04-openloco-v24.05.md:126]: Should we mention that due to this before drawing a window there is a whole bunch of logic rearranging widgets that now in theory doesn't need to happen.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**